lib/ansible/plugins/callback/default.py

```diff
@@ -145,13 +145,7 @@ def v2_playbook_on_play_start(self, play):
         self._display.banner(msg)
 
     def v2_on_file_diff(self, result):
-        if result._task.loop and 'results' in result._result:
-            for res in result._result['results']:
-                if 'diff' in res and res['diff'] and res.get('changed', False):
-                    diff = self._get_diff(res['diff'])
-                    if diff:
-                        self._display.display(diff)
-        elif 'diff' in result._result and result._result['diff'] and result._result.get('changed', False):
+        if 'diff' in result._result and result._result['diff'] and result._result.get('changed', False):
```

Comment:
Cannot confirm that. It's working as intended, with my other PR that was merged.

![image](https://cloud.githubusercontent.com/assets/345210/18676520/27a72648-7f56-11e6-856c-a79fc9a24f17.png)

![image](https://cloud.githubusercontent.com/assets/345210/18676633/8c5a737e-7f56-11e6-86e0-74488f1a8eeb.png)